Add course averages

SamZhang02 opened the pull request:
This PR adds the course averages using McGill Enhanced's crowdsourced sheet. The use was okay'ed by Koz and credits are given within the Class Average component.

The data is accessed using the Google Sheets API and a python script in `/tools`. It gets written as a Json in `/assets` and will be periodically updated (I am thinking every month or two). I chose to have to as a static asset instead of seeding it in the db because its a relative light file and it doesn't need to be updated by anyone else than the maintainers. This choice consistent with McGill Enhanced.

I'm still not sure if I can make the UI better tho, here is what it looks like right now:

![CleanShot 2024-05-02 at 12 35 03](https://github.com/terror/mcgill.courses/assets/112342947/f246620c-29be-416e-8168-ecc752a83107)

terror commented:
Nice work!

39bytes commented:
What about folding the averages to max 5-6 most recent then having a "show all" button to expand it? It feels like it can get a bit long if you show all of them by default
![image](https://github.com/terror/mcgill.courses/assets/47371088/08b9328c-f0de-4df7-8480-abd4b496d882)

(also, thoughts on making the font weight on the grades slightly more bold? i gave it `font-medium` in this screenshot)

39bytes commented:
Another useful thing I just thought of, it might be cool to put the instructor that taught during that term for each course average, can't think of a good way to fit it into the UI at the moment though, maybe just some really small text under the course term in the list?

SamZhang02 commented on client/src/components/CourseAverages.tsx:
```diff
@@ -0,0 +1,46 @@
+      <hr className='my-2 mt-5 w-full border border-neutral-200 dark:border-neutral-700' />
+
+      {averages
+        .sort((a, b) => compareTerms(a.term, b.term))
+        .reverse()
```
shit you're right

SamZhang02 commented:
> What about folding the averages to max 5-6 most recent then having a "show all" button to expand it? It feels like it can get a bit long if you show all of them by default. ![image](https://private-user-images.githubusercontent.com/47371088/327828676-08b9328c-f0de-4df7-8480-abd4b496d882.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3NTcyMjYsIm5iZiI6MTcxNDc1NjkyNiwicGF0aCI6Ii80NzM3MTA4OC8zMjc4Mjg2NzYtMDhiOTMyOGMtZjBkZS00ZGY3LTg0ODAtYWJkNGI0OTZkODgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDE3MjIwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2MmEyM2UzZTNjODNmYzgxNmRhODg4ZWZkZjAxNTdiZDIyNmY4NDM5NWFhMzk0MGE1OGM0YTE4NzgzYmI2ZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uYS47D9OSHf79rdkKa6nLxI1nUMqU16K2O2a7TPahcw)
> 
> (also, thoughts on making the font weight on the grades slightly more bold? i gave it `font-medium` in this screenshot)

yesss good idea i'll hop on this

> Another useful thing I just thought of, it might be cool to put the instructor that taught during that term for each course average, can't think of a good way to fit it into the UI at the moment though, maybe just some really small text under the course term in the list?

That's a really great idea yeah, I too struggle to think of the UI for now

39bytes commented:
Also one more thing, the text should scale better on mobile (too big)
![image](https://github.com/terror/mcgill.courses/assets/47371088/777fd21a-8883-4df7-b937-f6051ff0b99b)

SamZhang02 commented:
I addressed most comments except the one except the one about showing the prof of each semester, we may need some brainstorming for that hmmm

Maybe its better in a different PR? up to you guys

39bytes commented:
> I addressed most comments except the one except the one about showing the prof of each semester, we may need some brainstorming for that hmmm
> 
> Maybe its better in a different PR with some time for brainstorm? up to you guys

I just added it, what do you think?
![image](https://github.com/terror/mcgill.courses/assets/47371088/a908c1df-fd10-4a8c-82c0-bc84421d21c1)
